Frontend-unit-tests-guide.md

```diff
@@ -476,63 +548,75 @@ So, what you need to do is to change the order of beforeEach calls, as you can s
         TestBed.get(MyExampleService));
     }));
     ```
-- If you’re working with async on AngularJS and your tests don’t seem to run correctly, make sure you’re using `$apply` or `$digest` in the spec, as in this [example](https://github.com/oppia/oppia/blob/2e60d69d7b/core/templates/pages/topic-editor-page/services/topic-editor-state.service.spec.ts#L716-L720).
-
-## Services  
-Services are one of the most important features in the codebase. They contain logic that can be used across the codebase multiple times. There are three possible extensions for services:  
-- *.service.ts
-- *Factory.ts
-- *.factory.ts
-- *.tokenizer.ts  
-
-As a good first issue, all the services that need to be tested are listed in [issue #4057](https://github.com/oppia/oppia/issues/4057), grouped by the [complexity criteria](#how-to-choose-a-file-to-work-on).
-
-### AngularJS
-Use some files that are correctly following the testing patterns as reference:  
-- [current-interaction.service.spec.ts](https://github.com/oppia/oppia/blob/2e60d69d7b/core/templates/pages/exploration-player-page/services/current-interaction.service.spec.ts#L39)
-- [editable-exploration-backend-api.service.spec.ts](https://github.com/oppia/oppia/blob/2e60d69d7b/core/templates/domain/exploration/editable-exploration-backend-api.service.spec.ts#L30)
-- [improvement-task.service.spec.ts](https://github.com/oppia/oppia/blob/2e60d69d7b/core/templates/services/improvement-task.service.spec.ts#L29)
-
-### Angular 2+  
-Use some files that are correctly following the testing patterns as reference:  
-- [exploration-features-backend-api.service.spec.ts](https://github.com/oppia/oppia/blob/2e60d69d7b/core/templates/services/exploration-features-backend-api.service.spec.ts#L26)
-- [editability.service.spec.ts](https://github.com/oppia/oppia/blob/2e60d69d7b/core/templates/services/editability.service.spec.ts#L21)
-- [exploration-html-formatter.service.spec.ts](https://github.com/oppia/oppia/blob/2e60d69d7b/core/templates/services/exploration-html-formatter.service.spec.ts#L27)
-
-## Controllers
+
+* If you’re working with async on AngularJS and your tests don’t seem to run correctly, make sure you’re using `$apply` or `$digest` in the spec, as in this [example](https://github.com/oppia/oppia/blob/2e60d69d7b/core/templates/pages/topic-editor-page/services/topic-editor-state.service.spec.ts#L716-L720).
+
+## Testing services
+
+Services are one of the most important features in the codebase. They contain logic that can be used across the codebase multiple times. There are three possible extensions for services:
+
+* `*.service.ts`
+* `*Factory.ts`
+* `*.factory.ts`
+* `*.tokenizer.ts`
+
+As a good first issue, all the services that need to be tested are listed in [issue #4057](https://github.com/oppia/oppia/issues/4057).
+
+### Testing AngularJS services
+
+Use these files that are correctly following the testing patterns as reference:
```

Comment:
Here and below -- should it be "for reference" or "as references"?

Reply:
yep! done